src/bridgetree.rs

```diff
@@ -301,6 +302,19 @@ impl<H, const DEPTH: u8> Frontier<H, DEPTH> {
     pub fn position(&self) -> Option<Position> {
         self.frontier.as_ref().map(|f| f.position)
     }
+
+    pub fn dynamic_memory_usage(&self) -> usize {
+        size_of::<Option<NonEmptyFrontier<H>>>()
+            + self.frontier.as_ref().map_or(0, |f| {
+                2 * size_of::<usize>() + f.parents.capacity() * size_of::<H>()
+            })
```

Comment:
The first part is not dynamic memory usage, e.g. `self` could be on the stack. Only the caller knows whether this should be included in its own `dynamic_memory_usage` call, and it would figure that out with a call to `size_of::<Frontier<H, DEPTH>>()`.
```suggestion
        self.frontier.as_ref().map_or(0, |f| {
            2 * size_of::<usize>() + f.parents.capacity() * size_of::<H>()
        })
```
Technically this would also undercount if `H` itself allocated memory. I think it's fine to just document that limitation, since this struct only really makes sense with `H` types that are "simple".